runtime/parsing_functions.cpp

```diff
@@ -0,0 +1,325 @@
+#include "parsing_functions.h"
+
+/*
+ * Helper functions
+ */
+
+string clearQuotes(const string &str)
+{
+  if (str.starts_with(string("\'"))) {
+    return clearSingleQuotes(str);
+  }
+
+  if (str.starts_with(string("\""))) {
+    return clearDoubleQuotes(str);
+  }
+
+  return str;
+}
+
+string clearSingleQuotes(const string &str)
+{
+  if (str.empty()) {
+    return {};
+  }
+
+  if (str.starts_with(string("\'")) && str.ends_with(string("\'"))) {
+    return str.substr(1, str.size()-2);
+  }
+
+  return str;
+}
+
+string clearDoubleQuotes(const string &str)
+{
+  if (str.empty()) {
+    return {};
+  }
+
+  if (str.starts_with(string("\"")) && str.ends_with(string("\""))) {
+    return str.substr(1, str.size()-2);
+  }
+
+  return str;
+}
+
+string clearExtraSpaces(const string &str)
+{
+  if (str.empty()) {
+    return {};
+  }
+
+  string clear_str;
+
+  for (string::size_type i = 0; i < str.size(); i++) {
+    if (str[i] != ' ') {
+      clear_str.push_back(str[i]);
+    } else {
+      clear_str.push_back(' ');
+
+      while (str[i+1] == ' ') {
+        i++;
+      }
+    }
+  }
+
+  return clear_str;
+}
+
+string tolower(const string &str)
+{
+  if (str.empty()) {
+    return {};
+  }
+
+  string lower_str = {};
+
+  for (string::size_type i = 0; i < str.size(); i++) {
+    lower_str.push_back(tolower(str[i]));
+  }
+
+  return lower_str;
+}
+
+string trim(const string &str)
```

Comment:
I would suggest using `f$trim` function instead or any other equivalent.
It's better not to duplicate the functions with the same semantics.